### DISCUSSION TITLE
Is the expiration for recovery tokens configurable in Supabase cloud?

### DISCUSSION THREAD

**Discussion by chipilov** [supabase/supabase#6603] [Questions] [answered]
Supabase's GoTrue docs have this snippet:

```
MAILER_OTP_EXP - number

Controls the duration an email link or otp is valid for.
```

Also, the GoTrue code itself seems to refer to this config when verifying a recovery token: https://github.com/supabase/gotrue/blob/ca839e42ace1b5fbbcaa1a7573ba6edefa3e6866/api/verify.go#L468

However, I don't understand how to set/change the value of MAILER_OTP_EXP in a Supabase cloud project - is that possible? If not, what is the default? I don't see the ```MAILER_OTP_EXP``` option used in the docker file: https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml.

**Comment by GaryAustin1** [comment]
Another clue, but not definitive:
In https://github.com/supabase/gotrue/blob/master/conf/configuration.go	

if config.Mailer.OtpExp == 0 {
		config.Mailer.OtpExp = 86400 // 1 day
	}

**Reply by chipilov** [reply]
Thanks, indeed this seems to be the default - I tried using a recovery token after 24 hours and it did NOT work.

I guess the question then is how to modify it (if at all possible)?

**Comment by chipilov** [comment, marked as answer]
Just a got confirmation from the Supabase team that currently it is NOT possible to change the default 24-hour expiration of recovery tokens in cloud projects. They also mentioned that they plan to add the ability to change it soon.

In the meantime, if anyone needs to shorten the expiration time, they can run a cron job against the ```auth.users``` table and delete any tokens which have NOT been used for more than X hours/minutes. Extending the expiration time is also possible by modifying the ```recovery_token_sent_at``` column to a date in the future but this will lose the original value.

**Reply by cliveportman** [reply]
Urgh. Thanks for the response, though.
